components/backup-stream/src/utils.rs

```diff
@@ -536,6 +537,30 @@ pub fn with_record_read_throughput<T>(f: impl FnOnce() -> T) -> (T, u64) {
     (r, recorder.end())
 }
 
+/// test whether a key is in the range.
+/// end key is exclusive.
+/// empty end key means infinity.
+pub fn is_in_range(key: &[u8], range: (&[u8], &[u8])) -> bool {
+    match range {
+        (start, b"") => key >= start,
+        (start, end) => key >= start && key < end,
+    }
+}
+
+/// test whether two ranges overlapping.
+/// end key is exclusive.
+/// empty end key means infinity.
+pub fn is_overlapping(range: (&[u8], &[u8]), range2: (&[u8], &[u8])) -> bool {
+    let (x1, y1) = range;
+    let (x2, y2) = range2;
+    match (x1, y1, x2, y2) {
+        (_, b"", _, b"") => true,
+        (x1, b"", _, y2) => x1 < y2,
+        (_, x2, y1, b"") => y1 < x2,
```

Comment:
is it ok instead of `(_, y1, x2, b"") => x2 < y1` 👀

Reply:
Yes, but writing with that would be a little magical and not as symmetric as now...